week2/homework/bonus/bonus.js

```diff
@@ -0,0 +1,77 @@
+const program = require('commander');
+const fs = require('fs');
+
+program
+  .option('add <new todo>', 'Add new todo item', function(todoItem) {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      const todoList = JSON.parse(toDos);
+      todoList.push(todoItem);
+      const newList = JSON.stringify(todoList, null, 2);
+      console.log(newList);
+      return fs.writeFile('./bonus/list.json', newList, error => console.log(error));
+    });
+  })
+  .option('list', 'List all todo items', function() {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      console.log(toDos);
+    });
+  })
+  .option('reset', 'Removes all todo items', function() {
+    fs.writeFile('./bonus/list.json', '[]', error => console.log(error));
+  })
+  .option('remove <todo row>', 'Remove one todo item', function(todoItem) {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      const todoList = JSON.parse(toDos);
+      const filteredList = todoList.filter((elem, index) => index !== Number(todoItem) - 2);
+      const newList = JSON.stringify(filteredList, null, 2);
+      console.log(newList);
+      return fs.writeFile('./bonus/list.json', newList, error => console.log(error));
+    });
+  })
+  // The `update` takes only one argument, the second one is `undefined`.
```

Comment:
Looks like a known issue. It is suggested to use `program.args` https://github.com/tj/commander.js/issues/53#issuecomment-59824905

Reply:
I tried this one but it didn't work at all for some reason.

program
  .command('update')
  .usage('<row number> [new value...]')
  .description('update one todo item')
  .action(function(todoItem, newVal) {
    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
      if (error) {
        console.log(error);
      }
      const todoList = JSON.parse(toDos);
      const mappedList = todoList.map((elem, index) =>
        index === Number(todoItem) - 2 ? newVal : elem,
      );
      const newList = JSON.stringify(mappedList, null, 2);
      console.log(newList);
      return fs.writeFile('./bonus/list.json', newList, error => console.log(error));
    });
  });
`node bonus . update 2 "Do something else"` 
It does not make any changes and no messages at all in the `console`

Reply:
hmm, well no big deal, it is not your fault, it is an error in the library

Reply:
Great!